gcip/core/rule.py

```diff
@@ -22,6 +38,19 @@ class WhenStatement(Enum):
 
 
 class Rule:
+    """This module represents the Gitlab CI [rules](https://docs.gitlab.com/ee/ci/yaml/#rules) keyword.
+
+    [extended_summary]
+
+    Args:
+        if_statement (Optional[str], optional): The [rules:if clause](https://docs.gitlab.com/ee/ci/yaml/#when) which decides when
```

Comment:
We should consider either using `Optional[str]` or `str, optional` but both is  a bit redundant. 
Maybe we can have a talk about this later on. No change required.

Reply:
Yes.

That is the default output of my docs generator. Lets talk about this.